### PR TITLE
Persist the order of the comparison table rows

### PR DESCRIPTION
**Pull request by sroy3**
Fixes point [number 2 ](https://github.com/iterative/vscode-dvc/pull/2271#pullrequestreview-1089533215)

https://user-images.githubusercontent.com/3683420/187507629-b41641bb-6c0b-4e5b-8b5b-5d47d6d03e29.mov

### CODE REVIEW

File: extension/src/util/array.ts [hunk 1]
```diff
@@ -48,3 +48,25 @@ export const splitMatchedOrdered = <T>(values: T[], existingOrder: T[]) => {
 
 export const reorderListSubset = <T>(subset: T[], supersetOrder: T[]): T[] =>
   splitMatchedOrdered(subset, supersetOrder)[0]
+
+export const performOrderedUpdate = (
+  order: string[],
```
Comment (sroy3) [inline extension/src/util/array.ts:53]: Didn't make sense to move `performSimpleOrderedUpdate` and not this one as well.

**Comment by mattseddon** [comment]
Still seeing the same crash bug:

https://user-images.githubusercontent.com/37993418/187567253-b81765ce-271a-44b7-bcb9-1fdf9b7c6882.mov

I cannot recreate on `main`:

https://user-images.githubusercontent.com/37993418/187567459-7fcecc2b-caf2-480d-ae2d-3e30002f420c.mov

Will dig into it now.

**Comment by mattseddon** [comment]
In `webview/src/shared/components/dragDrop/DragDropContainer.tsx` & `<DragDropContainer>` you create a list of`wrappedItems`. Some of the items can now be undefined. One thing that seems to fix the issue is changing 
```
const { id } = draggable.props
```
to
```
 const id = draggable?.props?.id
```

After fixing that it looks like everything works as expected.

**Review by mattseddon — APPROVED** [review]

**Comment by sroy3** [comment]
> In `webview/src/shared/components/dragDrop/DragDropContainer.tsx` & `<DragDropContainer>` you create a list of`wrappedItems`. Some of the items can now be undefined. One thing that seems to fix the issue is changing
> 
> ```
> const { id } = draggable.props
> ```
> 
> to
> 
> ```
>  const id = draggable?.props?.id
> ```
> 
> After fixing that it looks like everything works as expected.

Fixed

**Pull request merged**